compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1144,23 +1114,20 @@ private STNode parseFuncDefOrFuncTypeDescRhs(STNode name, STNode metadata, STNod
                                                isObjectMember, isObjectTypeDesc, name);
             case EQUAL_TOKEN:
             case SEMICOLON_TOKEN:
-                endContext();
+                switchContext(ParserRuleContext.VAR_DECL_STMT);
```

Comment:
Why do we need this change?.

Reply:
After identifier(`function a;`) if it contains `equal` or `semicolon` token it should be variable declaration with function type.  isn't it? (It cannot be a function definition)

Reply:
Changed.